src/ecdsa.php

```diff
@@ -1,24 +1,59 @@
 <?php
 
 namespace EllipticCurve;
+
 use EllipticCurve\Signature;
+use EllipticCurve\Utils\Binary;
+use EllipticCurve\Utils\Integer;
 
 
 class Ecdsa {
     
-    public static function sign ($message, $privateKey) {
-        $signature = null;
-        openssl_sign($message, $signature, $privateKey->openSslPrivateKey, OPENSSL_ALGO_SHA256);
-        return new Signature($signature);
-    }
+    public static function sign ($message, $privateKey, $hashfunc="sha256"){
+        $byteMessage = hash($hashfunc, $message);
+        $numberMessage = Binary::intFromHex($byteMessage);
+        $curve = $privateKey->curve;
 
-    public static function verify ($message, $signature, $publicKey) {
-        $success = openssl_verify($message, $signature->toDer(), $publicKey->openSslPublicKey, OPENSSL_ALGO_SHA256);
-        if ($success == 1) {
-            return true;
+        
+        $r = Integer::toGmp(0);
+        $s = Integer::toGmp(0);
+        $recoveryId = null;
+        
+        while ($r == 0 or $s == 0){
+            $randNum = Integer::between(1, $curve->N - 1);
+            $randSignPoint = Math::multiply($curve->G, n:$randNum, A:$curve->A, P:$curve->P, N:$curve->N);
+            $r = $randSignPoint->x % $curve->N;
+            $s = (($numberMessage + $r * $privateKey->secret) * (Math::inv($randNum, $curve->N))) % $curve->N;
         }
+    
+        return new Signature(r:$r, s:$s, recoveryId:$recoveryId);
+    }
+
+    public static function verify ($message, $signature, $publicKey, $hashfunc="sha256") {
+
+        
```

Comment:
Deixar sem linhas em branco no início do método, pra deixar no padrão das outras libs